packages/tpetra/core/src/Tpetra_Details_makeColMap_def.hpp

```diff
@@ -590,8 +587,8 @@ makeColMap (Teuchos::RCP<const Tpetra::Map<LO, GO, NT>>& colMap,
   using device_t = typename NT::device_type;
   using exec_space = typename device_t::execution_space;
   using memory_space = typename device_t::memory_space;
-  using bitset_t = Kokkos::Bitset<device_t>;
-  using const_bitset_t = Kokkos::ConstBitset<device_t>;
+  using bitset_t = Kokkos::Bitset<exec_space>;
```

Comment:
What determines whether to use device_t or exec_space?

Reply:
device_t is an execution space/memory space pair. exec_space is just the execution space, so this bitset_t is now allocating into the execution space's default memory space (in this case of HIP, HIPSpace). This is required because most atomics in HIPHostPinnedSpace (like the bitwise OR needed by Bitset) cause the GPU to hang.

Reply:
BTW, this bitset is only used locally in this function, and only in kernels executing on the exec_space. So this won't cause any issues with type mismatches on other backends.

Reply:
It still feels a little weird to me to specify an execution space instead of a memory space. I would prefer the latter.